### DISCUSSION TITLE
Pinia error with Vue 2.7.x and jest

### DISCUSSION THREAD

**Discussion by jcmillett** [vuejs/pinia#1796] [Help and Questions]
### Reproduction

See reproduction steps

### Steps to reproduce the bug

I've used pinia with vue 2.7.x, and it's worked fine up until I try to write some unit tests for my store and components. I'm utilizing the latest versions of vue 2 (v2.7.14), pinia (v2.0.23), and jest (v29.3.1).

When I run my unit tests I get the following error: `vueDemi.effectScope is not a function`
![Screen Shot 2022-11-11 at 2 13 24 PM](https://user-images.githubusercontent.com/4006766/201431652-cf14f137-1e2b-445f-a84c-966fcb5b7122.png)

Store test file:
```
import { setActivePinia, createPinia } from 'pinia';
import { useMyStore } from '../myStore.js';

describe('helpDocsMappingsStore', () => {
	beforeEach(() => {
		setActivePinia(createPinia());
	});

	it('test', () => {
		useMyStore();
	});
});
```

It appears that the error occurs when `createPinia()` is called. I see a similar error when writing unit tests for vue components using `createTestingPinia()`  from `@pinia/testing`.

As best as I can tell, I believe this is a bug within pinia.

### Expected behavior

Should be able to run unit tests utilizing pinia, vue 2.7.x, and jest.

### Actual behavior

Getting an error stating that `vueDemi.effectScope is not a function`.

**Comment by posva** [comment]
You need to upgrade your Vue-Demi version. With npm, npm upgrade should do. You can also reinstall Vue-demi@latest

**Reply by jcmillett** [reply]
@posva Is vue-demi required in order to use Pinia? I didn't see anything in the docs that mentioned it. I don't have vue-demi listed as a dependency in my package.json, but it's already in my yarn.lock file, and set to the latest version v0.13.11. I explicitly added it to my package.json file with yarn, and I still get the same error.

**Comment by jaytdev** [comment]
I'm running into the same issue, and have yet to find a solution. Vue 2.7.14, Pinia 2.0.27, Jest 29.3.1. Tried installing vue-demi@0.13.11 and it didn't help. Tried `npx vue-demi-switch 2` and `npx vue-demi-switch 2.7` and `npx vue-demi-fix` and those didn't work either.

**Comment by jaytdev** [comment]
@posva Do you have any other suggestions on how to solve this? 🙏

**Comment by Vimm2** [comment]
Having the same issue in 2024.
vue 2 (v2.7.16), pinia (v2.2.4), jest (v29.7.0), @pinia/testing (v0.1.6) 

Calling createTestingPinia() throws the error:
`TypeError: vueDemi.effectScope is not a function`
